### PR TITLE
New Module: Truncate Long Links (#2113)

### LINKED ISSUE #2113: Option for post title max-height
specified in `em`?  maybe `4em` by default?  and `overflow: hidden; text-overflow: ellipsis`

https://www.reddit.com/r/Enhancement/comments/2z4d87/how_to_remove_all_titles_longer_than_one_line/

### PR DESCRIPTION
**Pull request by jkcchan**
Closes #2113

- Module to add CSS  `overflow: hidden; text-overflow: ellipsis; white-space: nowrap;`

### CODE REVIEW

**Comment by 13steinj** [comment]
Not really part of the project team, but IMO it shouldn't be it's own module, since it can't really be expanded upon greatly. I'd say add it to the ["Appearance"](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/lib/modules/betteReddit.js) (betteReddit) module.

**Comment by jkcchan** [comment]
@13steinj Thank you for the feedback! Will change it so that it's an option on that module.

File: lib/modules/betteReddit.js [hunk 1]
```diff
@@ -200,6 +205,9 @@ addModule('betteReddit', function(module, moduleID) {
 				default:
 					break;
 			}
+			if (module.options.truncateLongLinks.value) {
+				RESUtils.addCSS('overflow: hidden;text-overflow: ellipsis;white-space: nowrap;');
```
Comment (andytuba) [inline lib/modules/betteReddit.js]: There's a tidy new paradigm for these CSS only options: https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/lib/css/modules/_betteReddit.scss

Could you update to that style? Want a quick walkthrough on the pattern?

**Comment by andytuba** [comment]
Thanks for contributing these quality of life updates!

File: lib/modules/betteReddit.js (hunk 1 above)
Comment (jkcchan) [inline lib/modules/betteReddit.js]: @andytuba from what I understand, do you mean adding the SCSS selectors to the string that is passed in the addCSS function? i.e. `.link .title`?

Comment (andytuba) [inline lib/modules/betteReddit.js]: No, moving your changes into betteReddit.scss and wrapping them in `.res-betteReddit-truncateLongLinks { ... }`

Comment (jkcchan) [inline lib/modules/betteReddit.js]: @andytuba okay, will do. Is that all I have to do? Where would I put the boolean check?

Comment (13steinj) [inline lib/modules/betteReddit.js]: @jkcchan 

Just remove the boolean check. A body class is added to the page automatically when options are on. So you just remove the oolean check and add the following to the scss file.

    .res-betteReddit-truncateLongLinks {
        .link .title {
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: no-wrap;
        }
    }

Comment (erikdesjardins) [inline lib/modules/betteReddit.js]: Not quite, you need to add `bodyClass: true` to the option.

Comment (13steinj) [inline lib/modules/betteReddit.js]: whoops, my bad!

Comment (jkcchan) [inline lib/modules/betteReddit.js]: @13steinj @erikdesjardins Thank you for the guidance!

**Comment by erikdesjardins** [comment]
Merged in #2891.
Thanks @jkcchan!

**Pull request closed without merging**